pkg/repository/v1/sqlcv1/filters.sql

```diff
@@ -88,3 +88,24 @@ WHERE
     AND workflow_id = ANY(@workflowIds::UUID[])
 GROUP BY workflow_id
 ;
+
+-- name: ListFilters :many
+WITH inputs AS (
+    SELECT
+        UNNEST(COALESCE(sqlc.narg(workflowIds)::UUID[], '{}')) AS workflow_id,
+        UNNEST(COALESCE(sqlc.narg(scopes)::TEXT[], '{}')) AS scope
+), num_filter_inputs AS (
+    SELECT COUNT(*) AS ct
+    FROM inputs
+    WHERE workflow_id IS NOT NULL AND scope IS NOT NULL
+)
+
+SELECT f.*
+FROM v1_filter f
+CROSS JOIN num_filter_inputs n
+JOIN inputs i ON (n.ct = 0) OR (f.workflow_id, f.scope) = (i.workflow_id, i.scope)
```

Comment:
any thoughts on this? it felt pretty hacky but not sure if we have a different way that's preferred